final_task/rss_reader/links_collection.py

```diff
@@ -0,0 +1,27 @@
+from collections.abc import Collection
+
+
+class LinksCollection(Collection):
+
+    def __len__(self) -> int:
+        return len(self.items_dict)
+
+    def __contains__(self, __x: object) -> bool:
+        return __x in self.items_dict
+
+    def __iter__(self):
+        return self.items_dict.__iter__()
+
+    def __init__(self, initial=None):
+        self.items_dict = dict()
+        self._next_index_ = 1
+
+        if initial is not None:
```

Comment:
It is better to be consistent with typehint usage: either typehint everywhere or nowhere.